src/Cache/Engine/FileEngine.php

```diff
@@ -388,7 +388,7 @@ protected function _setKey($key, $createKey = false)
         if (!$createKey && !$path->isFile()) {
             return false;
         }
-        if (empty($this->_File) || $this->_File->getBasename() !== $key) {
+        if (null === $this->_File || $this->_File->getBasename() !== $key) {
```

Comment:
a) Please dont use YODA, === null always.
b) I agree that this change is not safe for master, as it can break with certain nullish values coming in that so far "just worked" (even if not intended by the code).